File: src/Python/examples/text_steps.py
from __future__ import annotations
from ..core.short_circuit import short_circuit

class TextSteps:
    @staticmethod
    def strip(s: str) -> str:
        return "" if s is None else s.strip()

    @staticmethod
    def normalize_whitespace(s: str) -> str:
        import re
        return re.sub(r"\s+", " ", s)

    @staticmethod
    def disallow_emoji(s: str) -> str:
        import re
        if re.search(r"[\u2600-\u26FF\u2700-\u27BF]", s or ""):
            raise ValueError("Emoji not allowed")
        return s

    @staticmethod
    def truncate_at_280(s: str) -> str:
        if s is not None and len(s) > 280:
            short_circuit(s[:280])
        return s

File: src/typescript/examples/text_steps.ts
import { shortCircuit } from "../core/short_circuit.js";

export class TextSteps {
  static strip(textValue: string | null | undefined): string {
    if (textValue == null) {
      return "";
    }
    const textString = String(textValue);
    return textString.trim();
  }

  static normalizeWhitespace(textValue: string): string {
    const textString = String(textValue);
    const result = textString.replace(/\s+/g, " ");
    return result;
  }

  static disallowEmoji(textValue: string): string {
    const textString = String(textValue);
    const hasEmoji = /[\u2600-\u26FF\u2700-\u27BF]/.test(textString);
    if (hasEmoji) {
      throw new Error("Emoji not allowed");
    }
    return textString;
  }

  static truncateAt280(textValue: string): string {
    const textString = String(textValue);
    if (textString.length > 280) {
      shortCircuit(textString.slice(0, 280));
    }
    return textString;
  }
}
